### PR TITLE
TSK-1543: Throw expressive error when REST parameter cannot be converted

### PR DESCRIPTION
**Pull request by tge20**
---
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [x] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → TSK-XXX: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

**Comment by tge20** [comment]
https://sonarcloud.io/dashboard?branch=TSK-1543&id=tge20_taskana

File: rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/TaskanaRestExceptionHandler.java [hunk 1]
```diff
@@ -33,113 +35,113 @@
...
+  protected ResponseEntity<Object> handleBindException(
+      BindException ex, HttpHeaders headers, HttpStatus status, WebRequest request) {
+    return buildResponse(ex, request, HttpStatus.BAD_REQUEST);
+  }
+
   @ExceptionHandler(Exception.class)
-  protected ResponseEntity<TaskanaErrorData> handleGeneralException(Exception ex, WebRequest req) {
+  protected ResponseEntity<Object> handleGeneralException(Exception ex, WebRequest req) {
     return buildResponse(ex, req, HttpStatus.BAD_REQUEST);
   }
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/TaskanaRestExceptionHandler.java:137]: Did you test that this is working?

File: rest/taskana-rest-spring/src/test/java/pro/taskana/workbasket/rest/WorkbasketControllerIntTest.java [hunk 2]
```diff
@@ -311,4 +312,22 @@ void should_ThrowException_When_ProvidingInvalidFilterParams() {
...
+        () -> {
+          TEMPLATE.exchange(url, HttpMethod.GET, auth, WORKBASKET_SUMMARY_PAGE_MODEL_TYPE);
+        };
+
+    assertThatThrownBy(httpCall)
+        .isInstanceOf(BadRequest.class)
+        .hasMessageContaining(
+            "\"status\":400,\"error\":\"BAD_REQUEST\",\"exception\":\"org.springframework.web."
+                + "method.annotation.ModelAttributeMethodProcessor$1\",\"message\":\"org."
+                + "springframework");
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/workbasket/rest/WorkbasketControllerIntTest.java:331]: Better: Please verify that part of the message which says that "GROU" is invalid

Comment (tge20) [reply rest/taskana-rest-spring/src/test/java/pro/taskana/workbasket/rest/WorkbasketControllerIntTest.java:331]: That's what I was attempting to do. Unfortunately I realized that only the first part of the message is displayed / checked without that information.

File: rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/TaskanaRestExceptionHandler.java (hunk 1 above)
Comment (tge20) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/TaskanaRestExceptionHandler.java:137]: Yes, it works as we predicted for Taskana Exceptions which are not mentioned explicitly before.

**Review by mustaphazorgati — APPROVED** [review]

**Pull request merged**